### ISSUE TITLE
SSD1325 Controller

### ISSUE THREAD

**Issue by Tulsa2008** [javl/image2cpp#8]
03/13/17

I see your software was written for the 1306 controller.  Does your software support the SSD1325 controller??

I have not been able to get the program to generate a file that works.  All I get is random junk on the screen.  

You test screen does work using your example program after a few changes.
I had to add:
 "#include <Adafruit_SSD1325.h>"
  Adafruit_SSD1325 display(OLED_DC, OLED_RESET, OLED_CS);

And change the pin assignments for DC, RST and CS.  

I  am using the hardware SPI communication protocol on a Mega 2560 processor. 

Any comment you might have will be greatly appreciated.
Ken

**Comment by javl** [comment]
As I don't have any other displays than the ones with a 1306 controller I don't have a way to test. I don't know what kind of pixel order they use, but we might be able to figure it out.  

First: what do you mean by "your test screen does work"?

**Comment by Tulsa2008** [comment]
03/14/17

First off, thanks for your quick response.  It is greatly appreciated.

When I first loaded your program, it had a test piece of code that showed an "A", a less than symbol, and some lines on it.  That code worked. 

I did think of something last night.  I have noticed that the code is extremely sensitive to the height and width declaration. The picture I am using has a bit of an odd size "89 x 60".  My original thoughts were that as long as the size of the declaration matched the "Canvas Size" all would be well regardless of the numbers.  Am I correct in this assumption??

I am using a black background and the Horizontal polarity.  I have done it both ways.  Same result.  

I have included the picture for you to try.  It is just a copy of the ADA Purple that has been cropped
down to size.

Thanks again for your assistance

Ken 




  
  ----- Original Message -----

**Comment by Tulsa2008** [comment]
03/14/17

Jasper:
I did some playing.

The order of pixel assignment is as I would have expected it.  If you have a row of data points 16 bytes long (as in your data base) with all bytes defined as FF, and you define the display as a 128 * 1 you will get a line drawn across the screen.  

If you setup a line with bytes 1-8 = ff and 9-16 = 00 and a line with bytes 1-8 = 00 and 9-16 = ff, define the display as 64 * 4.  You will get two lines 64 pixels long separated by two blank lines.  

Now something I haven't ask,  I assumed that the graphic file you  were looking for was a BMP file.  Is this true??  My graphic file is also a color graphic.  Is this a problem??


Thanks in advance for your assistance

Ken



  ----- Original Message -----

**Comment by Tulsa2008** [comment]
03/14/17

Jasper:
I got it to work by changing the canvas size to 128*64 even though the picture is only 89*60.

Again, Thanks for your response

Ken

  ----- Original Message -----

**Comment by javl** [comment]
Because of the pixels being divided into these chunks a small error will quickly result into noise. 
Great that you got it to work, and thanks for just trying a bunch of things yourself. 

To still answer your BMP/color question: that doesn't matter. Any conventional image type will do and the colors will be changed to black and white based on a threshold you can set.

**Comment by Tulsa2008** [comment]
03/15/17

Jasper:
Thanks a lot for your help.  

I think I found that the width of the image as defined in:
 "display.drawBitmap(X, Y, Bitmap2, 128, 54, WHITE)",  always has to be at least 128 regardless of the size of the image. The width can be larger than 128 but not smaller. I went as large as 232 and the image displayed as expected.  

The height does not have to be 64, it can be larger or smaller.

I actually made the image significantly larger than 128 x 64 (232 * 128) and found that I could move the image up and down on the display (panning).  I was pleasantly surprised to find that the x, y location definitions could be positive or negative.  

I have not played with the threshold to date,  I'll have to play with that to see what it does.  

Again, thanks for your help.  

Ken


  ----- Original Message -----
